examples/groth16-verifier/methods/guest/src/bin/groth16_verifier.rs

```diff
@@ -12,22 +12,16 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#![no_main]
+use risc0_groth16::{Fr, Seal, Verifier, VerifyingKey};
+use risc0_zkvm::{guest::env, sha::Digestible};
 
-use risc0_groth16::Verifier;
-use risc0_zkvm::{guest::env, sha::Digest};
-use sha2::{Digest as _, Sha256};
+pub fn main() {
+    let (seal, public_inputs, verifying_key): (Seal, Vec<Fr>, VerifyingKey) = env::read();
 
-risc0_zkvm::guest::entry!(main);
+    Verifier::new(&seal, &public_inputs, &verifying_key)
```

Comment:
Now, with this change we are almost doubling the required cycles to verify a groth16 proof in the guest, specifically from ~490 on main VS 980 on this branch. Are we aware of this?

Reply:
I saw that, and I should have called it out in particular. My thought here is that whether it's 1B vs 500M cycles, it's kind of both in the realm of "too expensive to run on a regular basis". My suggestion here is that we make this refactor now, and revisit this later, especially when we have an accelerator for pairings. My goal here is forwards compatibility with those kinds of changes, and we could reintroduce a extra info like the key preparation host-side when we revisit this to improve performance.